src/test/java/PA165/language_school_manager/Dao/LectureDaoTest.java

```diff
@@ -0,0 +1,192 @@
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ContextConfiguration(classes = ApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class LectureDaoTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceContext
+    public EntityManager em;
+
+    @Autowired
+    private LectureDao lectureDao;
+
+    @Autowired
+    private CourseDao courseDao;
+
+    @Autowired
+    private LecturerDAO lecturerDAO;
+
+    private Lecture lecture1;
+
+    @BeforeMethod
+    public void setup() {
+        lecture1 = new Lecture();
+        lecture1.setTopic("jamesBond");
+        lecture1.setTime(LocalDateTime.now().plusDays(2));
+
+        Lecturer lecturer = new Lecturer();
+        lecturer.setLastName("Bond");
+        lecturer.setFirstName("James");
+        lecturer.setMiddleName("007");
+        lecturer.setNativeSpeaker(true);
+        lecturer.addLanguage(Language.ENGLISH);
+        lecturer.addLanguage(Language.ITALIAN);
+        lecturerDAO.create(lecturer);
+
+        Course course = new Course();
+        course.setName("agent007");
+        course.setLanguage(Language.ENGLISH);
+        course.setProficiencyLevel(ProficiencyLevel.A1);
+        course.addLecture(lecture1);
+        courseDao.create(course);
+
+
+        lecture1.setCourse(course);
+        lecture1.setLecturer(lecturer);
+
+        lectureDao.create(lecture1);
+    }
+
+    @Test
+    public void createLecture() {
+        assertThat(lectureDao.findById(lecture1.getId())).isNotNull();
+        assertThat(lecture1.getTopic()).isEqualTo("jamesBond");
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void createNullLecture() {
+        lectureDao.create(null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void createLectureWithNullDate() {
+        lecture1.setTime(null);
+        lectureDao.create(lecture1);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void createLectureWithDateBeforeNow() {
+        lecture1.setTime(LocalDateTime.now().minusMinutes(10));
+        lectureDao.create(lecture1);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void createLectureWithNullCourse() {
+        lecture1.setCourse(null);
+        lectureDao.create(lecture1);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void createLectureWithNullLecturer() {
+        lecture1.setLecturer(null);
+        lectureDao.create(lecture1);
+    }
+
+    @Test
+    public void findLectureById() {
+        Lecture lecture = lectureDao.findById(lecture1.getId());
+        assertThat(lecture.getTopic()).isNotNull();
+        assertThat(lecture.getTopic()).isEqualTo("jamesBond");
```

Comment:
je to mozne takto napisat.
assertThat(lecture.getTopic()).isNotNull().isEqualTo("jamesBond");